platforms/core-execution/build-cache/src/main/java/org/gradle/caching/internal/controller/BuildCacheController.java

```diff
@@ -28,7 +30,10 @@
 
 /**
  * High-level controller for build-cache operations; can load and store {@link CacheableEntity}s with a given {@link BuildCacheKey}.
+ *
+ * <p>Note that some implementations are mutable and may change behavior over the lifetime of a build.</p>
  */
+@ServiceScope(Scope.Gradle.class)
 public interface BuildCacheController extends Closeable {
```

Comment:
It's unrelated to this PR, but when this code is shared with Maven, these annotations will be missing. (Not yet, because they are currently part of `:java-language-extensions`, but they should move to `:service-registry-api` and that library we are not going to share.)